File: C++/binary-tree-level-order-traversal.cpp
/**
 *  * Definition for a binary tree node.
 *   * struct TreeNode {
 *    *     int val;
 *     *     TreeNode *left;
 *      *     TreeNode *right;
 *       *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 *        * };
 *         */
class Solution {
private:
    void traverse(TreeNode *root, int level, std::vector<std::vector<int>> &ans) {
        if(root) {
            if(level>=ans.size()) ans.push_back(std::vector<int>());
            ans[level].push_back(root->val);
            traverse(root->left, level+1, ans);
            traverse(root->right, level+1, ans);
        }
    }
public:
    vector<vector<int>> levelOrder(TreeNode *root) {
        std::vector<std::vector<int>> ans;
        traverse(root, 0, ans);
        return ans;
    }
};

File: Python/binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param {TreeNode} root
    # @return {integer[][]}
    def levelOrder(self, root):
        def traverse(root, lv):
            if root:
                if lv >= len(ans): ans.append([])
                ans[lv].append(root.val)
                traverse(root.left, lv+1)
                traverse(root.right, lv+1)

        ans = []
        traverse(root, 0)
            
        return ans
